lib/web/app/transport.go

```diff
@@ -388,7 +388,7 @@ func (t *transport) DialContext(ctx context.Context, _, _ string) (conn net.Conn
 
 	// eliminate any servers from the head of the list that were unreachable
 	t.mu.Lock()
```

Comment:
What if some other concurrent call to `DialContext` already eliminated some servers from the head of `t.c.servers`? Is there some external synchronization to prevent that?

Reply:
I can't tell if `t.c.servers` is ever externally modified; if not, I guess that a length check would work?

Is `t.c.servers` ordered in some way? Does it have some uniqueness guarantee?